src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -614,6 +622,9 @@ public bool Remove(TKey key)
                         {
                             entries[last].next = entries[i].next;
                         }
+
+                        value = entries[i].value;
```

Comment:
@danmosemsft @mikedn Thanks for the review comments. I will work on getting the performance data. This review gives me a close and significant insight into working of C# and how to think differently when wearing a C# vs C++ hat.